R-packages/evalcast/R/scale_by_forecaster.R

```diff
@@ -0,0 +1,61 @@
+#' Scale error measures based on those of a particular forecaster.
+#'
+#' @param score_card score_card like that returned by `evaluate_predictions()`
+#' @param score_cols vector of column names in `score_card` to normalize
+#' @param base_forecaster_name name of forecaster in `score_card$forecaster` column by whose error
+#'   values the remaining forecasters' errors will be scaled
+#' @param numeric_err_cols vector of column names in `score_card` that contain numeric error
+#'   measures. Elements of `numeric_err_cols` that are not in `score_cols` will be dropped from the
+#'   final output to avoid a mix of scaled and unscaled measures.
+#'
+#' @return
+#' @export
+scale_by_forecaster <- function(score_card,
+                                score_cols,
+                                base_forecaster_name, 
+                                numeric_err_cols = c("ae", "wis")) {
```

Comment:
I don't think I like this function definition. The user needs to name the columns to normalize and then also name all the numeric scores (or hope those in the score_card match the default). I understand the goal of dropping unscaled columns, but I think convenience should supercede.

If `score_card` has the right class, then we should know which columns it has, and the rest will be `numeric_err_cols`. We can then drop any that aren't in `score_cols`. I think the 4th argument should be something like `id_vars = NULL`. If score card has the right class, we ignore it, otherwise, if someone passes in a random tibble with some identifying vars and some `score_cols` they want scaled, then we require them to list those things.

Reply:
> If score_card has the right class, then we should know which columns it has

`eval_predictions` can take a `side_truth` and arbitrary scoring functions, so I don't agree that we can know which columns will be present in a score card. The longer term solution is probably to have all objects of class `score_card` track which columns are identifiers vs. score columns.

Reply:
If the df comes out of `evaluate_predictions()`, we expect it to have some identifying columns and the `actual`. Anything else is either scores or extraneous. I'd rather the default behavior return the necessary id columns and the scaled scores (maybe the actuals) and dump the rest. If the user wants more columns in the return object, then they can `left_join()` the input and the results. If they provide something that's different than the `score_card` class, then let them specify. That way it "works" out of the box without requiring too much input.

Reply:
changed to make the columns to keep an argument with a default value.